### DISCUSSION TITLE
Blog Plugin: Recent Posts list to integrate with other pages

### DISCUSSION THREAD

**Discussion by athshean** [squidfunk/mkdocs-material#7720] [Ideas]
It would be great to have some way of generating recent posts that can be placed on the home page or in a footer or something. If this could be a partial template, it might even be something we can customise into integrating with cards or something for our sites.

What I'm thinking is designing a home page where we have the general landing content and then below that a card grid with up to 6 recent posts. As it stands, my workaround is manually writing cards after updating the blog. But having some automatic way of doing this and integrating rendering a blog list on an existing page might be handing.

**Comment by squidfunk** [comment]
This is already tracked in #5074.

**Comment by kamilkrzyskow** [comment]
In the [mkdocs-nype](https://github.com/nypesap/mkdocs-nype) theme, various plugins were introduced to extend the Blog plugin.
The [latest_blog_posts](https://github.com/nypesap/mkdocs-nype/tree/main/mkdocs_nype/plugins/latest_blog_posts) works on the home page and injects Markdown which works with the grid cards of the Material theme, it does use a [special token in the HTML](https://github.com/nypesap/nypesap.github.io/blob/main/docs/index.md?plain=1#L26).

You can see the latest blog posts plugin in action on the [Nype brand site](https://npe.cm/), or the [Fiori Tracker site](https://fioritracker.org/).

Other blog plugins also include:
- [custom_blog_categorization](https://github.com/nypesap/mkdocs-nype/tree/main/mkdocs_nype/plugins/custom_blog_categorization), which adds another layer of categorization for posts, can be seen [here with the Industries](https://npe.cm/projects/) example
- [similar_blog_posts](https://github.com/nypesap/mkdocs-nype/tree/main/mkdocs_nype/plugins/similar_blog_posts), which adds a sort of footer with a list of other posts based on the categories, can be seen [here in "SAP Fiori setup transparency"](https://fioritracker.org/usecases/SPS03/transparency/)

Please note that the `mkdocs-nype` theme should currently be used as reference to make your own solutions (the licence is MIT). The current "alpha/beta" stage assumes that there will be breaking changes without notice, and I'm not entirely sure if the "stable" release would assure compatibility with general projects outside of the Nype domain.

For example the latest addition to the blog at Nype is the ["cards display"](https://github.com/nypesap/mkdocs-nype/blob/main/mkdocs_nype/extensions/material.py#L145) and it's injected into the blog plugin at runtime, and likely won't be implemented as a separate plugin. It can be seen [here](https://fioritracker.org/usecases/), where both exploded and normal cards are used.

There is also no documentation yet, but once that is done, I will make a Show and Tell post here :v:
